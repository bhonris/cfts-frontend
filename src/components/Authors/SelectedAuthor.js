import React from 'react'

const SelectedAuthor = (props) => {
  const displayString = props.author === ""?
  "No Author Selected":
    `Selected Author: ${props.author}`
  return <label>{displayString}</label>
}

export default SelectedAuthor