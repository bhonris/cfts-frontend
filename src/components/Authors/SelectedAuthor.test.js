import { shallow } from 'enzyme'
import toJSON from "enzyme-to-json";
import SelectedAuthor from './SelectedAuthor';

describe("Test for Seelcted Author", ()=>{
  it("should match blank author snapshot", ()=>{
    const component = shallow(<SelectedAuthor author={""}/>);
    expect(toJSON(component)).toMatchSnapshot();
  })

  it("should match with author name snapshot", ()=>{
    const component = shallow(<SelectedAuthor author={"Friedrich Nietzsche"}/>);
    const text = component.find("label").text();
    expect(text).toEqual("Selected Author: Friedrich Nietzsche");

  })
})