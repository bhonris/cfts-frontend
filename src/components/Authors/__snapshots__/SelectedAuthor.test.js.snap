// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Seelcted Author should match blank author snapshot 1`] = `
<label>
  No Author Selected
</label>
`;

